integrations/clickup/client.py

```diff
@@ -0,0 +1,231 @@
+import asyncio
+import time
+from typing import Any, AsyncGenerator, Optional, List, Dict
+from httpx import HTTPStatusError, Timeout
+from loguru import logger
+from port_ocean.utils import http_async_client
+from port_ocean.utils.cache import cache_iterator_result
+
+TEAM_OBJECT = "__team"
+WEBHOOK_EVENTS = [
+    "taskCreated",
+    "taskUpdated",
+    "taskDeleted",
+    "listCreated",
+    "listUpdated",
+    "listDeleted",
+]
+
+MINIMUM_LIMIT_REMAINING = 20  # This limit minus the Semaphore count will be the lower threshold for rate limit in the header
+DEFAULT_SLEEP_TIME = 30  # The Default wait time when limit is hit
+SEMAPHORE = asyncio.BoundedSemaphore(10)  # The number of concurrent requests allowed
```

Comment:
set semaphore value as a constant.. e.g MAX_CONCURRENT_TASK.
Also, why 10 ?

Reply:
This has been done. The semaphore as been adjusted to a bigger default and recalculated based on the rate limit that the clickup account has.